### PR TITLE
Simplified require conditions for AMB bridge

### PR DESCRIPTION
**Pull request by k1rill-fedoseev**
It seems to me, that some of the `requires` for AMB bridge can be simplified, doing so may save some gas during execution.

### CODE REVIEW

File: contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol [hunk 1]
```diff
@@ -13,7 +13,7 @@ contract BasicAMB is BasicBridge {
         address _owner
     ) external onlyRelevantSender returns (bool) {
         require(!isInitialized());
-        require(_validatorContract != address(0) && AddressUtils.isContract(_validatorContract));
```
Comment (akolotov) [inline contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol]: There are similar checks in `erc20_to_native/HomeBridgeErcToNative.sol` and `RewardableBridge.sol`. Consider to re-factor code there as well.

Comment (k1rill-fedoseev) [reply contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol]: In https://github.com/poanetwork/tokenbridge-contracts/blob/c85aeb737bfd684b20b99520bd632ba4b8866583/contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol#L140 we also want to allow to set `0x00..00` as a `blockReward` address, so here we cannot simplify this check.

Similarly, in https://github.com/poanetwork/tokenbridge-contracts/blob/c85aeb737bfd684b20b99520bd632ba4b8866583/contracts/upgradeable_contracts/RewardableBridge.sol#L59-L62 we want to allow set `0x00..00` as a fee manager contract address.

Comment (k1rill-fedoseev) [reply contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol]: By the way, is it supposed to be so?
In `erc20_to_erc20/HomeBridgeErcToErc.sol`, `erc20_to_native/HomeBridgeErcToNative.sol`, `native_to_erc20/ForeignBridgeNativeToErc.sol` and `native_to_erc20/HomeBridgeNativeToErc.sol` we allow `_feeManager` to be a contract address only, as shown here https://github.com/poanetwork/tokenbridge-contracts/blob/c85aeb737bfd684b20b99520bd632ba4b8866583/contracts/upgradeable_contracts/erc20_to_erc20/HomeBridgeErcToErc.sol#L94, but later, in https://github.com/poanetwork/tokenbridge-contracts/blob/c85aeb737bfd684b20b99520bd632ba4b8866583/contracts/upgradeable_contracts/RewardableBridge.sol#L59-L62 we can also use `address(0)` as a fee manager address.

Should we refactor this checks to make them consistent?

Comment (akolotov) [reply contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol]: > we also want to allow to set `0x00..00` as a `blockReward` address, so here we cannot simplify this check.

Sorry, you are right. The checks are not the same

Comment (akolotov) [reply contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol]: > we can also use `address(0)` as a fee manager address.
> Should we refactor this checks to make them consistent?

It was done to provide ability to disable a Fee Manager without upgrade of the bridge contract. So, the current implementation is correct.

**Pull request merged**